Sources/BraveWallet/Crypto/Stores/SwapTokenStore.swift

```diff
@@ -502,7 +509,147 @@ public class SwapTokenStore: ObservableObject {
     guard status == .success, amountToSend > allowanceValue else { return }  // no problem with its allowance
     self.state = .lowAllowance(spenderAddress)
   }
-
+  
+  @MainActor private func fetchSolPriceQuote(
+    base: SwapParamsBase,
+    swapParams: BraveWallet.SwapParams,
+    network: BraveWallet.NetworkInfo
+  ) async {
+    guard base == .perSellAsset else {
+      return // entering buy amount is disabled for Solana swap
+    }
+    let jupiterQuoteParams: BraveWallet.JupiterQuoteParams = .init(
+      inputMint: swapParams.sellToken,
+      outputMint: swapParams.buyToken,
+      amount: swapParams.sellAmount,
+      slippagePercentage: swapParams.slippagePercentage
+    )
+    self.updatingPriceQuote = true
+    let (jupiterQuote, swapErrorResponse, _) = await swapService.jupiterQuote(jupiterQuoteParams)
+    defer { self.updatingPriceQuote = false }
+    if let jupiterQuote {
+      await self.handleSolPriceQuoteResponse(jupiterQuote, base: base)
+    } else if let swapErrorResponse {
+      // check balance first because error can be caused by insufficient balance
+      if let sellTokenBalance = self.selectedFromTokenBalance,
+         let sellAmountValue = BDouble(self.sellAmount.normalizedDecimals),
+         sellTokenBalance < sellAmountValue {
+        self.state = .error(Strings.Wallet.insufficientBalance)
+        return
+      }
```

Comment:
what kind of `swapErrorResponse` will it be when it is caused by insufficient balance? can we just use the  `message`? also wondering how `swapService` would know the balance.

Reply:
We had the same logic for Ethereum swap so I included it here in the error flow. I guess the selected token balance could have changed while fetching the price quote and we'd want to show insufficient balance before we show insufficient liquidity